os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -457,6 +455,18 @@ void up_assert(const uint8_t *filename, int lineno)
 
 	abort_mode = true;
 
+	uint32_t assert_pc;
+
+	/* Extract the PC value of instruction which caused the abort/assert */
+
+	if (g_assertpc) {
+		/* g_assertpc is set only during up_assert syscall or during faults*/
+		assert_pc = (uint32_t)g_assertpc;
+		g_assertpc = 0x0; /* reset */
+	} else {
+		assert_pc = (uint32_t) __builtin_return_address(0);
```

Comment:
As discussed, `__builtin_return_address` works only with some compilers, such as gcc.
Please add condition or replace it to `ARCH_GET_RET_ADDRESS` which is in mm.h

Reply:
I have replaced builtin_return_address with ARCH_GET_RET_ADDRESS

Reply:
Thanks for modification.
But as my test, we should call `ARCH_GET_RET_ADDRESS` at the first of the function, without calling any other function.
But in this code, there is `board_led_on`.
Please move the calling of `ARCH_GET_RET_ADDRESS` before calling `board_led_on`.

Reply:
I moved ARCH_GET_RET_ADDRESS to the start of the function. Thanks for pointing it out.